operator/standalone/controller.go

```diff
@@ -89,15 +89,16 @@ func (r *PostgresStandaloneReconciler) CreateDeployment(ctx context.Context, ins
 	return reconcile.Result{RequeueAfter: 10 * time.Second}, err
 }
 
-// Delete prepares the given instance for deletion.
-func (r *PostgresStandaloneReconciler) Delete(ctx context.Context, instance *v1alpha1.PostgresqlStandalone) (reconcile.Result, error) {
-	// we don't need to delete it by ourselves, since the deletion timestamp is already set.
-	// Just remove all finalizers
+// DeleteDeployment prepares the given instance for deletion.
+func (r *PostgresStandaloneReconciler) DeleteDeployment(ctx context.Context, instance *v1alpha1.PostgresqlStandalone) (reconcile.Result, error) {
 	log := ctrl.LoggerFrom(ctx)
-	controllerutil.RemoveFinalizer(instance, finalizer)
-	log.Info("Deleting")
-	err := r.client.Update(ctx, instance)
-	return reconcile.Result{}, err
+	d := NewDeleteStandalonePipeline(r.client, instance)
+	log.Info("Deleting instance")
+	err := d.RunPipeline(ctx)
+	if err != nil {
+		return reconcile.Result{RequeueAfter: 10 * time.Second}, err
+	}
+	return reconcile.Result{}, nil
```

Comment:
I think our resource has to be reconciled multiple times until it's really deleted.
Consider this:
- Deletion pipeline issues the Delete() on all relevant resources and returns without error.
- Those resources have finalizers and it takes a moment to cleanup -> we reconcile again in a few seconds.
- Once the helm release is truly gone, we can remove our finalizer.

With my previous suggestions to wait until `Release` is deleted, the pipeline returns `nil` even if the steps to delete the namespace and removing the finalizer didn't run due to the `pipeline.If` condition.
That means the result of the pipeline is `nil`, but we need to reconcile again in any case.
```suggestion
	err := d.RunPipeline(ctx)
	return reconcile.Result{RequeueAfter: 10 * time.Second}, err
```
(side note if you have concerns about infinte reconciles: At https://github.com/vshn/appcat-service-postgresql/blob/e23a3f0d704ea1f86ce19978631eebfeb63c4a95/operator/standalone/controller.go#L55-L58 we ensure that we don't reconcile anymore if our resource is gone after removing the last finalizer).

Reply:
I will recheck the code and will add tests to make sure we cover all cases!